### PR TITLE
Compilation with Java 6 and above

### PR DESCRIPTION
**Pull request by jbee**
Changed or added generics that would otherwise cause compilation errors with newer Java versions.
The generics as they are now should be fine with both Java 6 and above.

There a still **a lot** of raw types...

Please add more/other reviews that you think fit.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/util/SortingUtil.java [hunk 1]
```diff
@@ -128,9 +128,9 @@ public int compare(Map.Entry entry1, Map.Entry entry2) {
         };
     }
 
-    public static Comparator<Map.Entry> newComparator(final PagingPredicate pagingPredicate) {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/util/SortingUtil.java]: This class is my main point of worry. Officially hazelcast.util is private, so there should not be any problems if an external user would be running in a potential incompatibility problem..

**Comment by pveentjer** [comment]
@jbee can you rebase?

**Comment by jbee** [comment]
For some reason I did not get email notifications on this issue. Just discovered by chance that there has been feedback. So I was not ignoring anything intentionally. 

Now, regarding your concerns. I had already given up on this one - I just forgot to close it. So if you ask me we just close it now.

**Comment by mmedenjak** [comment]
@jbee can you rebase and merge?

**Pull request merged**